### ISSUE TITLE
Search for words individually (with implicit logical and)

### ISSUE THREAD

**Issue by h-h-h-h** [davidhealey/waistline#472]
Use case: I want to find the salmon dish from ThatOne brand with many products without diving into the details of how exactly they named it. In the food list, I enter "salmon ThatOne", but it doesn't yield any results, because the space is taken literally, just like the letters.

I'd like the behavior to be like in the search function of many internet forums: Every word is individually searched for in the item to be presented, and every word must occur in the item.

If this shouldn't be the only behavior in your opinion, maybe add it (or the opposite) as a checkbox option to the list that appears when you tap the filter icon in the search textbox.

---

BTW: It'd also be nice if items that are *not* the results of the search would be displayed after the results in normal sort order, e.g., with dimmer text color and after separating vertical space. This would allow you to more often select multiple items in one go instead of repeatedly visiting the food list. Could also be a checkbox option.

Even better would be a gathering feature: Search, select, gather (like `+` or so next to `✔`); search again, select, `✔`.

**Comment by davidhealey** [comment]
Are you referring to searching OFF or searching within the results of your own local foodlist?

**Comment by h-h-h-h** [comment]
Searching in my own local foodlist. How can you even search in OFF?

**Comment by davidhealey** [comment]
> Searching in my own local foodlist.

This should be possible because this search is handled by Waistline. Searching OFF though is handled by their servers so Waistline has little control over how the search terms are handled.

> How can you even search in OFF?

Press enter/return after writing your search query (assuming you're connected to the internet).

**Comment by h-h-h-h** [comment]
> > How can you even search in OFF?
>
> Press enter/return after writing your search query (assuming you're connected to the internet).

That's quite a hidden feature.

**Comment by davidhealey** [comment]
> That's quite a hidden feature.

Most search functions require you to hit enter before it searches.

**Comment by h-h-h-h** [comment]
> Most search functions require you to hit enter before it searches.

You wouldn't expect *two* kinds of searches, though, when it works without hitting enter.

**Comment by davidhealey** [comment]
On my phone the enter/return button has a search icon when typing in the search box, that should make it clear that pressing that button will trigger a search of some kind, additional to the real time filtering. Does your enter button not show a search icon?

**Comment by h-h-h-h** [comment]
Yes, I have a search icon on the enter button. In the browser, in the Google search textbox etc., there's a volatile overlay, though, that's distinctly different from the webpage area or the search engine results. And it's something helping you with what to search for, not already final results that can be utilized.

I never paid attention to the enter button in this instance, because, in other parts of the app, it's also meaningless/without function: In the food editor, e.g., I have a "Next" button for the upper textboxes, but for the bottom-most one, I have a "Go" button ("Los" in German) that doesn't do anything.

**Comment by davidhealey** [comment]
Well I'm not suggesting the implementations between a browser and Waistline are like for like. The point is that they both have a text box that serves two purposes depending on if the return button is pressed or not after a query has been entered.
